cli-toolchain/graphql.md

```diff
@@ -1739,11 +1754,288 @@ mutation CreateProject {
 
-Likewise, you can make a simple one-to-many connection as follows:
+Likewise, you can make a simple one-to-many connection as follows for a post that has many comments:
+
+**Has Many**
+
+```graphql
+type Post @model {
+  id: ID!
+  title: String!
+  comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+
+type Comment @model
+  @key(name: "byPost", fields: ["postID", "content"]) {
+  id: ID!
+  postID: ID!
+  content: String!
+}
+```
+
+Note how a one-to-many connection needs a @key that allows comments to be queried by the postID and the connection uses this key to get all comments whose postID is the id of the post was called on.
+After it's transformed, you can create comments and query the connected Post as follows:
+
+```graphql
+mutation CreatePost {
+  createPost(input: { id: "a-post-id", title: "Post Title" } ) {
+    id
+    title
+  }
+}
+
+mutation CreateCommentOnPost {
+  createComment(input: { id: "a-comment-id", content: "A comment", postID: "a-post-id"}) {
+    id
+    content
+  }
+}
+```
+
+And you can query a Post with its comments as follows:
+
+```graphql
+query getPost {
+  getPost(id: "a-post-id") {
+    id
+    title
+    comments {
+      items {
+        id
+        content
+      }
+    }
+  }
+}
+```
+
+**Belongs To**
+
+You can make a connection bi-directional by adding a many-to-one connection to types that already have a one-to-many connection. In this case we add a connection from Comment to Post since each comment belongs to a post:
+
+```graphql
+type Post @model {
+  id: ID!
+  title: String!
+  comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+
+type Comment @model
+  @key(name: "byPost", fields: ["postID", "content"]) {
+  id: ID!
+  postID: ID!
+  content: String!
+  post: Post @connection(fields: ["postID"])
+}
+```
+
+After it's transformed, you can create comments with a post as follows:
+
+```graphql
+mutation CreatePost {
+  createPost(input: { id: "a-post-id", title: "Post Title" } ) {
+    id
+    title
+  }
+}
+
+mutation CreateCommentOnPost1 {
+  createComment(input: { id: "a-comment-id-1", content: "A comment #1", postID: "a-post-id"}) {
+    id
+    content
+  }
+}
+
+mutation CreateCommentOnPost2 {
+  createComment(input: { id: "a-comment-id-2", content: "A comment #2", postID: "a-post-id"}) {
+    id
+    content
+  }
+}
+```
+
+And you can query a Comment with its Post, then all Comments of that Post by navigating the connection:
+
+```graphql
+query GetCommentWithPostAndComments {
+  getComment( id: "a-comment-id-1" ) {
+    id
+    content
+    post {
+      id
+      title
+      comments {
+        items {
+          id
+          content
+        }
+      }
+    }
+  }
+}
+```
+
+**Many-To-Many Connections**
+
+You can implement many to many yourself using two 1-M @connections and a joining @model. For example:
```

Comment:
You also need an `@key` for this use case. I know it's called out below but that's after this documentation section.